panther_manager/include/panther_manager/plugins/shutdown_host.hpp

```diff
@@ -0,0 +1,258 @@
+    state_(ShutdownHostState::IDLE)
+  {
+  }
+
+  ~ShutdownHost() {}
+
+  void call()
+  {
+    switch (state_) {
+      case ShutdownHostState::IDLE:
+        if (!is_available()) {
+          state_ = ShutdownHostState::SKIPPED;
+          break;
+        }
+
+        try {
+          request_shutdown();
+        } catch (std::runtime_error err) {
+          state_ = ShutdownHostState::FAILURE;
+          failure_reason_ = err.what();
+          break;
+        }
+        state_ = ShutdownHostState::COMMAND_EXECUTED;
+        break;
+
+      case ShutdownHostState::COMMAND_EXECUTED:
+        try {
+          if (update_response()) {
+            break;
+          }
+        } catch (std::runtime_error err) {
+          state_ = ShutdownHostState::FAILURE;
+          failure_reason_ = err.what();
+          break;
+        }
+        state_ = ShutdownHostState::RESPONSE_RECEIVED;
+        break;
+
+      case ShutdownHostState::RESPONSE_RECEIVED:
+        state_ = ShutdownHostState::PINGING;
+        break;
+      case ShutdownHostState::PINGING:
+        if (ping_for_success_ ? !is_available() : true) {
+          state_ = ShutdownHostState::SUCCESS;
+          break;
+        }
+        if (timeout_exceeded()) {
+          state_ = ShutdownHostState::FAILURE;
+          failure_reason_ = "Timeout exceeded";
+        }
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  bool is_available() const
+  {
+    return system(("ping -c 1 -w 1 " + ip_ + " > /dev/null").c_str()) == 0;
+  }
+
+  void close_connection()
+  {
+    if (ssh_channel_is_closed(channel_)) {
+      return;
+    }
+
+    ssh_channel_send_eof(channel_);
+    ssh_channel_close(channel_);
+    ssh_channel_free(channel_);
+    ssh_disconnect(session_);
+    ssh_free(session_);
+  }
+
+  int get_port() const { return port_; }
+
+  std::string get_ip() const { return ip_; }
+
+  std::string get_user() const { return user_; }
+
+  std::string get_command() const { return command_; }
+
+  std::string get_error() const { return failure_reason_; }
+
+  std::string get_response() const { return output_; }
+
+  ShutdownHostState get_state() const { return state_; }
+
+  bool operator==(const ShutdownHost & other) const { return hash_ == other.hash_; }
+
+  bool operator!=(const ShutdownHost & other) const { return hash_ != other.hash_; }
+
+  bool operator<(const ShutdownHost & other) const { return hash_ < other.hash_; }
+
+private:
+  const std::string ip_;
+  const std::string user_;
+  const std::string command_;
+  const std::size_t hash_;
+  const int port_;
+  const bool ping_for_success_;
+  const float timeout_;
+
+  char buffer_[1024];
+  const int verbosity_ = SSH_LOG_NOLOG;
+  int nbytes_;
+  std::string output_;
+  std::string failure_reason_;
+  ros::Time command_time_;
+  ShutdownHostState state_;
+
+  ssh_session session_;
+  ssh_channel channel_;
+
+  void request_shutdown()
+  {
+    ssh_execute_command(ip_.c_str(), user_.c_str(), command_.c_str(), port_);
```

Comment:
```suggestion
    ssh_execute_command(command_);
```

Reply:
The function might accept only command since it has access to the host parameters